Keypad/Keypad.js

```diff
@@ -0,0 +1,327 @@
+/* eslint-disable react/jsx-no-bind */
+
+/**
+ * Provides Agate-themed keypad components and behaviors. Used to display a sequence of numbers and buttons, like a keyboard.
+ *
+ * @example
+ * <Keypad />
+ *
+ * @module agate/Keypad
+ * @exports Keypad
+ * @exports KeypadBase
+ */
+
+import {adaptEvent, forward, handle} from '@enact/core/handle';
+import kind from '@enact/core/kind';
+import Layout, {Cell} from '@enact/ui/Layout';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Button from '../Button';
+
+import $L from '../internal/$L';
+
+import css from './Keypad.module.less';
+
+const KEY_LIST = [
+	{text: '1', subtext: ''},
+	{text: '2', subtext: 'abc'},
+	{text: '3', subtext: 'def'},
+	{text: '4', subtext: 'ghi'},
+	{text: '5', subtext: 'jkl'},
+	{text: '6', subtext: 'mno'},
+	{text: '7', subtext: 'pqrs'},
+	{text: '8', subtext: 'tuv'},
+	{text: '9', subtext: 'wxyz'},
+	{text: '*', subtext: ''},
+	{text: '0', subtext: ''},
+	{text: '#', subtext: ''},
+	{text: 'phone', subtext: ''},
+	{text: 'arrowleftturn', subtext: ''}
+];
+
+/**
+ * Renders an Agate-styled Key button.
+ *
+ * @class Key
+ * @memberof agate/Keypad
+ * @ui
+ * @public
+ */
+const Key = kind({
+	name: 'Key',
+
+	propTypes: /** @lends agate/Keypad.Key.prototype */ {
+		/**
+		 * Called when this button is clicked. Includes the 'key' key in its event payload to let the clicker know what was clicked inside their callback.
+		 *
+		 * @type {Function}
+		 * @param {Object} event
+		 * @public
+		 */
+		onKeyButtonClick: PropTypes.func,
+
+		/**
+		 * Text displayed below the number/icon/symbol on the key.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		subtext: PropTypes.string,
+
+		/**
+		 * Text displayed in the center of the key.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		text: PropTypes.string
+	},
+
+	styles: {
+		css,
+		className: 'key'
+	},
+
+	computed: {
+		textComponent: ({text}) => {
+			return (text != null && text !== '') ? <span className={css.text}>{text}</span> : null;
+		},
+		subtextComponent: ({subtext}) => {
+			return (subtext != null && subtext !== '') ? <span className={css.subtext}>{subtext}</span> : null;
+		}
```

Comment:
Let's move these from `computed` down into render and simplify them like the following:
```html
<Button
	{...rest}
	css={css}
	icon={children}
	size="large"
>
	{(text || text === 0) ? <span className={css.text}>{text}</span> : null}
	{(subtext || subtext === 0) ? <span className={css.subtext}>{subtext}</span> : null}
</Button>
```
It looks like we really only care about _truthy_ (and also 0), so we can make the conditional much simpler.